libsel4vmmplatsupport/src/arch/arm/guest_image.c

```diff
@@ -148,49 +155,48 @@ static int guest_write_address(vm_t *vm, uintptr_t paddr, void *vaddr, size_t si
     return 0;
 }
 
+static size_t get_file_size(const char *image_name)
+{
+    FILE *file = fopen(image_name, "r");
+    if (!file) {
+        return 0;
+    }
+    fseek(file, 0, SEEK_END);
+    size_t size = ftell(file);
+    fseek(file, 0, SEEK_SET);
+
+    fclose(file);
+
+    return size;
+}
+
 static int load_image(vm_t *vm, const char *image_name, uintptr_t load_addr,  size_t *resulting_image_size)
 {
-    int fd;
-    size_t len;
+    int fd[1];
```

Comment:
What cant this remain `int fd;`, seems we make it a 1-element array now for no reason?

Reply:
You're right, it can just stay as an int. While I was putting this patch together I was playing around with the idea of passing some other information to the callback, but it wasn't necessary.